File: frontend/src/app/models/Question.ts
import { Answer } from "./Answer";
import { Option } from "./Option";


export class Question{
  
  id?: number;
  statement?: string;
  label?: string;
  position?: number;
  typeId?: string;
  subCategoryId?: number;
  categoryId?: number;
  options: Option[] = [];
  isOptional?: boolean;
  answers: Answer[] = []

  constructor (
    {id, statement, label, position, subCategoryId, categoryId, typeId, isOptional}: 
    {id?: number, statement?: string, label?: string, position?: number, subCategoryId?: number, categoryId?: number, typeId?: string, isOptional?: boolean}
    ) {
      this.id = id;
      this.statement = statement;
      this.label = label;
      this.position = position;
      this.subCategoryId = subCategoryId;
      this.categoryId = categoryId;
      this.typeId = typeId;
      this.isOptional = isOptional;
    }
}

File: backend/CoreCuestionariosOIJ/src/CuestionariosEntidades/Models/Question.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CuestionariosEntidades.Models
{
    public class Question
    {
        public int? Id { get; set; }
        public string? Statement { get; set; } = string.Empty;
        public string? Label { get; set; } = string.Empty;
        public int? Position { get; set; }
        public int? CategoryId { get; set; }
        public int? SubCategoryId { get; set; }
        public int? QuestionnaireId { get; set; }
        public int? TypeId { get; set; }
        public bool? IsOptional { get; set; }

        // EF Relation
        public IEnumerable<Answer>? Answers { get; set; }
        public Questionnaire? Questionnaire;
        public Category? Category;
        public SubCategory? SubCategory;
        public QuestionType? QuestionType;
        public IEnumerable<Option>? Options { get; set; }

    }
}
